clevertap-core/src/main/java/com/clevertap/android/sdk/userEventLogs/UserEventLogDAOImpl.kt

```diff
@@ -0,0 +1,361 @@
+package com.clevertap.android.sdk.userEventLogs
+
+import android.content.ContentValues
+import android.database.Cursor
+import android.database.sqlite.SQLiteDatabase
+import androidx.annotation.WorkerThread
+import com.clevertap.android.sdk.Logger
+import com.clevertap.android.sdk.Utils
+import com.clevertap.android.sdk.db.Column
+import com.clevertap.android.sdk.db.DBAdapter.Companion.DB_OUT_OF_MEMORY_ERROR
+import com.clevertap.android.sdk.db.DBAdapter.Companion.DB_UPDATE_ERROR
+import com.clevertap.android.sdk.db.DBAdapter.Companion.NOT_ENOUGH_SPACE_LOG
+import com.clevertap.android.sdk.db.DatabaseHelper
+import com.clevertap.android.sdk.db.Table
+
+internal class UserEventLogDAOImpl(
+    private val db: DatabaseHelper,
+    private val logger: Logger,
+    private val table: Table
+) : UserEventLogDAO {
```

Comment:
`getColumnIndexOrThrow` can be done once, not sure if it will improve perf, do consider

Reply:
`getColumnIndexOrThrow` internally uses a Map of <ColumnName, index> which is a cache so won't be any perf impact. Keeping as is.

```
private Map<String, Integer> mColumnNameMap;
Integer i = mColumnNameMap.get(columnName);
        if (i != null) {
            return i.intValue();
        } else {
            return -1;
        }
```